src/main/kotlin/cacophony/diagnostics/Diagnostics.kt

```diff
@@ -0,0 +1,15 @@
+package cacophony.diagnostics
+
+import cacophony.utils.Location
+
+interface Diagnostics {
+    fun report(
+        message: DiagnosticMessage,
+        location: Location,
+    ) = report(message, location to location)
```

Comment:
Input a nicer `toString` on a single `Location`. Now, we wouldn't use it. So maybe change `locationRangeToString` to call `locationToString` given range of length 1.